src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -62,8 +62,23 @@ public Type[] GetStartupTypes()
         public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
         {
             string binPath;
-            if (_extensionBundleManager.IsExtensionBundleConfigured())
+            var functionMetadataCollection = _functionMetadataProvider.GetFunctionMetadata(forceRefresh: true);
```

Comment:
We need to call this regardless of bundle configuration because we need the metadata provider to drop the cache everything the host restart.